### DISCUSSION TITLE
Model list is incomplete

### DISCUSSION THREAD

**Discussion by tqangxl** [open-webui/open-webui#2659] [General]
# Bug Report

## Description
Model list is incomplete， set Model  list num <200?

![image](https://github.com/open-webui/open-webui/assets/9669944/a4cb17d4-3ba1-4dd4-b951-f3853fa7e334)
![image](https://github.com/open-webui/open-webui/assets/9669944/2f2ff5e9-251f-42d8-b9d7-4a3c77dca609)
![image](https://github.com/open-webui/open-webui/assets/9669944/2bd8961d-792e-4ba5-bc4a-3d97ccf7bbd5)

**Bug Summary:**
[Provide a brief but clear summary of the bug]

Steps to Reproduce:
run open-webui1\backend\start_windows.bat
chose  ollama model from list 

Expected Behavior:
[Describe what you expected to happen.]

Actual Behavior:
[Describe what actually happened.]

Environment
image

Operating System: [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]

Windows 11 专业版
版本 23H2
操作系统版本 22631.3593
体验 Windows Feature Experience Pack 1000.22700.1003.0

Browser (if applicable): [e.g., Chrome 100.0, Firefox 98.0]
Microsoft Edge 125.0.2535.67 (正式版本) (64 bit)

Reproduction Details
Confirmation:

[ yes] I have read and followed all the instructions provided in the README.md.
[yes ] I am on the latest version of both Open WebUI and Ollama.
[yes ] I have included the browser console logs.
Logs and Screenshots

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]
Backend config: {status: true, name: 'Open WebUI', version: '0.1.125', auth: true, default_locale: 'en-US', …}admin_export_enabled: trueauth: truedefault_locale: "en-US"default_models: nulldefault_prompt_suggestions: Array(7)0: {title: Array(2), content: "Help me study vocabulary: write a sentence for me …e blank, and I'll try to pick the correct option."}1: {title: Array(2), content: "What are 5 creative things I could do with my kids…o throw them away, but it's also so much clutter."}2: {title: Array(2), content: 'Tell me a random fun fact about the Roman Empire'}3: {title: Array(2), content: "Show me a code snippet of a website's sticky header in CSS and JavaScript."}4: {title: Array(2), content: "Explain options trading in simple terms if I'm familiar with buying and selling stocks."}5: {title: Array(2), content: 'Could you start by asking me about instances when …and then give me some suggestions to overcome it?'}6: {title: Array(2), content: 'Check the following sentence for grammar and clari…adability while maintaining its original meaning.'}length: 7[[Prototype]]: Array(0)at: ƒ at()concat: ƒ concat()constructor: ƒ Array()copyWithin: ƒ copyWithin()entries: ƒ entries()every: ƒ every()fill: ƒ fill()filter: ƒ filter()find: ƒ find()findIndex: ƒ findIndex()findLast: ƒ findLast()findLastIndex: ƒ findLastIndex()flat: ƒ flat()flatMap: ƒ flatMap()forEach: ƒ forEach()includes: ƒ includes()indexOf: ƒ indexOf()join: ƒ join()keys: ƒ keys()lastIndexOf: ƒ lastIndexOf()length: 0map: ƒ map()pop: ƒ pop()push: ƒ push()reduce: ƒ reduce()reduceRight: ƒ reduceRight()reverse: ƒ reverse()shift: ƒ shift()slice: ƒ slice()some: ƒ some()sort: ƒ sort()splice: ƒ splice()toLocaleString: ƒ toLocaleString()toReversed: ƒ toReversed()toSorted: ƒ toSorted()toSpliced: ƒ toSpliced()toString: ƒ toString()unshift: ƒ unshift()values: ƒ values()with: ƒ with()Symbol(Symbol.iterator): ƒ values()Symbol(Symbol.unscopables): {at: true, copyWithin: true, entries: true, fill: true, find: true, …}[[Prototype]]: Objectimages: falsename: "Open WebUI"status: truetrusted_header_auth: falseversion: "0.1.125"[[Prototype]]: Objectconstructor: ƒ Object()hasOwnProperty: ƒ hasOwnProperty()isPrototypeOf: ƒ isPrototypeOf()propertyIsEnumerable: ƒ propertyIsEnumerable()toLocaleString: ƒ toLocaleString()toString: ƒ toString()valueOf: ƒ valueOf()__defineGetter__: ƒ __defineGetter__()__defineSetter__: ƒ __defineSetter__()__lookupGetter__: ƒ __lookupGetter__()__lookupSetter__: ƒ __lookupSetter__()__proto__: (...)get __proto__: ƒ __proto__()set __proto__: ƒ __proto__()
6.20336259.js:5 {id: '', title: '新聊天', models: Array(1), options: {…}, messages: Array(1), …}history: currentId: "cc75ef30-2cf4-400a-b59c-fb4c405fb175"messages: {cc75ef30-2cf4-400a-b59c-fb4c405fb175: {…}}[[Prototype]]: Objectid: ""messages: [{…}]models: ['phi3:14b-medium-128k-instruct-q4_K_S']0: "phi3:14b-medium-128k-instruct-q4_K_S"length: 1[[Prototype]]: Array(0)at: ƒ at()concat: ƒ concat()constructor: ƒ Array()copyWithin: ƒ copyWithin()entries: ƒ entries()every: ƒ every()fill: ƒ fill()filter: ƒ filter()find: ƒ find()findIndex: ƒ findIndex()findLast: ƒ findLast()findLastIndex: ƒ findLastIndex()flat: ƒ flat()flatMap: ƒ flatMap()forEach: ƒ forEach()includes: ƒ includes()indexOf: ƒ indexOf()join: ƒ join()keys: ƒ keys()lastIndexOf: ƒ lastIndexOf()length: 0map: ƒ map()pop: ƒ pop()push: ƒ push()reduce: ƒ reduce()reduceRight: ƒ reduceRight()reverse: ƒ reverse()shift: ƒ shift()slice: ƒ slice()some: ƒ some()sort: ƒ sort()splice: ƒ splice()toLocaleString: ƒ toLocaleString()toReversed: ƒ toReversed()toSorted: ƒ toSorted()toSpliced: ƒ toSpliced()toString: ƒ toString()unshift: ƒ unshift()values: ƒ values()with: ƒ with()Symbol(Symbol.iterator): ƒ values()Symbol(Symbol.unscopables): {at: true, copyWithin: true, entries: true, fill: true, find: true, …}[[Prototype]]: Objectoptions: {}tags: []timestamp: 1716835186043title: "新聊天"[[Prototype]]: Object
**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

Installation Method
stall from Open WebUI Github Repo

Requirements 📦
🐰 [Node.js](https://nodejs.org/en) >= 20.10
🐍 [Python](https://python.org/) >= 3.11

PS D:\Lib\Dev\AI\open-webui> python
Python 3.11.9 (tags/v3.11.9:de54cf5, Apr 2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.

PS D:\Lib\Dev\AI\open-webui> node -v
v20.13.1
PS D:\Lib\Dev\AI\open-webui>

Build and Install 🛠️
Run the following commands to install:
git clone https://github.com/open-webui/open-webui.git
cd open-webui/

Copying required .env file
cp -RPp .env.example .env

Building Frontend Using Node
npm i
npm run build

Serving Frontend with the Backend
cd ./backend
pip install -r requirements.txt -U
run open-webui1\backend\start_windows.bat

**Comment by pascal456** [comment]
I experienced the same and was confused at first. But if you look closely you will see that the model you mention actually exists two times!

From the `ollama list` command you can see that the hash for the model exists two times. I do not 100% know how the inner workings of ollama are regarding that but my assumption is, that this is solely for practical reasons. Having downloaded the model once means it can be reused for other names or tags (which is called `NAME` in the `ollama list` command output) but the has (called `ID` in output list) is the same.
